realm/realm-library/src/main/java/io/realm/RealmBaseSchema.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.util.Map;
+import java.util.Set;
+
+import io.realm.internal.ColumnIndices;
+import io.realm.internal.ColumnInfo;
+import io.realm.internal.RealmProxyMediator;
+import io.realm.internal.Table;
+
+
+/**
+ * Class for interacting with the Realm schema using a dynamic API. This makes it possible
+ * to add, delete and change the classes in the Realm.
+ * <p>
+ * All changes must happen inside a write transaction for the particular Realm.
+ *
+ * @see RealmMigration
+ */
+public abstract class RealmBaseSchema {
```

Comment:
@beeender I don't exactly understand why this works the way it does, either.  I suspect it is because it is in transition.  I suspect that, over time, functionality will be moved to the `RealmNativeSchema` class.  Eventually the RealmSchema class will shrivel up and die.  When it does, the abstract base class can go away too.